cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/WorkflowDriver.java

```diff
@@ -505,13 +507,24 @@ private DatasetProperties addLocalDatasetProperty(DatasetProperties properties)
   }
 
   private void createLocalDatasets() throws IOException, DatasetManagementException {
-    for (Map.Entry<String, String> entry : datasetFramework.getDatasetNameMapping().entrySet()) {
-      String localInstanceName = entry.getValue();
-      DatasetId instanceId = new DatasetId(workflowRunId.getNamespace(), localInstanceName);
-      DatasetCreationSpec instanceSpec = workflowSpec.getLocalDatasetSpecs().get(entry.getKey());
+    for (final Map.Entry<String, String> entry : datasetFramework.getDatasetNameMapping().entrySet()) {
+      final String localInstanceName = entry.getValue();
+      final DatasetId instanceId = new DatasetId(workflowRunId.getNamespace(), localInstanceName);
+      final DatasetCreationSpec instanceSpec = workflowSpec.getLocalDatasetSpecs().get(entry.getKey());
       LOG.debug("Adding Workflow local dataset instance: {}", localInstanceName);
-      datasetFramework.addInstance(instanceSpec.getTypeName(), instanceId,
-                                   addLocalDatasetProperty(instanceSpec.getProperties()));
+
+      Retries.callWithRetries(new Retries.Callable<Void, IOException>() {
+        @Override
+        public Void call() throws IOException {
+          try {
+            datasetFramework.addInstance(instanceSpec.getTypeName(), instanceId,
+                                         addLocalDatasetProperty(instanceSpec.getProperties()));
+          } catch (DatasetManagementException e) {
```

Comment:
Sorry, I didn't realise that `DatasetManagementException` was a child of `RetryableException`.